File: output/csharp/Cat/CatNodes/CatNodesRecord.cs
using Nest.Internal;
using Nest.Cat;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Cat {

	public class CatNodesRecord : ICatRecord {
		
		[DataMember(Name="build")]
		public string Build { get; set; }


		[DataMember(Name="completion_size")]
		public string CompletionSize { get; set; }


		[DataMember(Name="cpu")]
		public string Cpu { get; set; }


		[DataMember(Name="disk_available")]
		public string DiskAvailable { get; set; }


		[DataMember(Name="fielddata_evictions")]
		public string FielddataEvictions { get; set; }


		[DataMember(Name="fielddata_memory")]
		public string FielddataMemory { get; set; }


		[DataMember(Name="file_descriptor_current")]
		public int FileDescriptorCurrent { get; set; }


		[DataMember(Name="file_descriptor_max")]
		public int FileDescriptorMax { get; set; }


		[DataMember(Name="file_descriptor_percent")]
		public int FileDescriptorPercent { get; set; }


		[DataMember(Name="filter_cache_evictions")]
		public string FilterCacheEvictions { get; set; }


		[DataMember(Name="filter_cache_memory")]
		public string FilterCacheMemory { get; set; }


		[DataMember(Name="flush_total")]
		public string FlushTotal { get; set; }


		[DataMember(Name="flush_total_time")]
		public string FlushTotalTime { get; set; }


		[DataMember(Name="get_current")]
		public string GetCurrent { get; set; }


		[DataMember(Name="get_exists_time")]
		public string GetExistsTime { get; set; }


		[DataMember(Name="get_exists_total")]
		public string GetExistsTotal { get; set; }


		[DataMember(Name="get_missing_time")]
		public string GetMissingTime { get; set; }


		[DataMember(Name="get_missing_total")]
		public string GetMissingTotal { get; set; }


		[DataMember(Name="get_time")]
		public string GetTime { get; set; }


		[DataMember(Name="get_total")]
		public string GetTotal { get; set; }


		[DataMember(Name="heap_current")]
		public string HeapCurrent { get; set; }


		[DataMember(Name="heap_max")]
		public string HeapMax { get; set; }


		[DataMember(Name="heap_percent")]
		public string HeapPercent { get; set; }


		[DataMember(Name="id_cache_memory")]
		public string IdCacheMemory { get; set; }


		[DataMember(Name="indexing_delete_current")]
		public string IndexingDeleteCurrent { get; set; }


		[DataMember(Name="indexing_delete_time")]
		public string IndexingDeleteTime { get; set; }


		[DataMember(Name="indexing_delete_total")]
		public string IndexingDeleteTotal { get; set; }


		[DataMember(Name="indexing_index_current")]
		public string IndexingIndexCurrent { get; set; }


		[DataMember(Name="indexing_index_time")]
		public string IndexingIndexTime { get; set; }


		[DataMember(Name="indexing_index_total")]
		public string IndexingIndexTotal { get; set; }


		[DataMember(Name="ip")]
		public string Ip { get; set; }


		[DataMember(Name="jdk")]
		public string Jdk { get; set; }


		[DataMember(Name="load_15m")]
		public string Load_15m { get; set; }


		[DataMember(Name="load_5m")]
		public string Load_5m { get; set; }


		[DataMember(Name="load_1m")]
		public string Load_1m { get; set; }


		[DataMember(Name="master")]
		public string Master { get; set; }


		[DataMember(Name="merges_current")]
		public string MergesCurrent { get; set; }


		[DataMember(Name="merges_current_docs")]
		public string MergesCurrentDocs { get; set; }


		[DataMember(Name="merges_current_size")]
		public string MergesCurrentSize { get; set; }


		[DataMember(Name="merges_total")]
		public string MergesTotal { get; set; }


		[DataMember(Name="merges_total_docs")]
		public string MergesTotalDocs { get; set; }


		[DataMember(Name="merges_total_time")]
		public string MergesTotalTime { get; set; }


		[DataMember(Name="name")]
		public string Name { get; set; }


		[DataMember(Name="node_id")]
		public string NodeId { get; set; }


		[DataMember(Name="node_role")]
		public string NodeRole { get; set; }


		[DataMember(Name="percolate_current")]
		public string PercolateCurrent { get; set; }


		[DataMember(Name="percolate_memory")]
		public string PercolateMemory { get; set; }


		[DataMember(Name="percolate_queries")]
		public string PercolateQueries { get; set; }


		[DataMember(Name="percolate_time")]
		public string PercolateTime { get; set; }


		[DataMember(Name="percolate_total")]
		public string PercolateTotal { get; set; }


		[DataMember(Name="pid")]
		public string Pid { get; set; }


		[DataMember(Name="port")]
		public string Port { get; set; }


		[DataMember(Name="ram_current")]
		public string RamCurrent { get; set; }


		[DataMember(Name="ram_max")]
		public string RamMax { get; set; }


		[DataMember(Name="ram_percent")]
		public string RamPercent { get; set; }


		[DataMember(Name="refresh_time")]
		public string RefreshTime { get; set; }


		[DataMember(Name="refresh_total")]
		public string RefreshTotal { get; set; }


		[DataMember(Name="search_fetch_current")]
		public string SearchFetchCurrent { get; set; }


		[DataMember(Name="search_fetch_time")]
		public string SearchFetchTime { get; set; }


		[DataMember(Name="search_fetch_total")]
		public string SearchFetchTotal { get; set; }


		[DataMember(Name="search_open_contexts")]
		public string SearchOpenContexts { get; set; }


		[DataMember(Name="search_query_current")]
		public string SearchQueryCurrent { get; set; }


		[DataMember(Name="search_query_time")]
		public string SearchQueryTime { get; set; }


		[DataMember(Name="search_query_total")]
		public string SearchQueryTotal { get; set; }


		[DataMember(Name="segments_count")]
		public string SegmentsCount { get; set; }


		[DataMember(Name="segments_index_writer_max_memory")]
		public string SegmentsIndexWriterMaxMemory { get; set; }


		[DataMember(Name="segments_index_writer_memory")]
		public string SegmentsIndexWriterMemory { get; set; }


		[DataMember(Name="segments_memory")]
		public string SegmentsMemory { get; set; }


		[DataMember(Name="segments_version_map_memory")]
		public string SegmentsVersionMapMemory { get; set; }


		[DataMember(Name="uptime")]
		public string Uptime { get; set; }


		[DataMember(Name="version")]
		public string Version { get; set; }

	}
}

File: specification/cat/cat_nodes/CatNodesRecord.ts
/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { ByteSize, Id, Name, Percentage, Type, VersionString } from '../../__common/common'

export class CatNodesRecord {
  /**
   * unique node id
   * @aliases nodeId
   */
  'id'?: Id
  /**
   * process id
   * @aliases p
   */
  'pid'?: string
  /**
   * ip address
   * @aliases i
   */
  'ip'?: string
  /**
   * bound transport port
   * @aliases po
   */
  'port'?: string
  /**
   * bound http address
   * @aliases http
   */
  'http_address'?: string

  /**
   * es version
   * @aliases v
   */
  'version'?: VersionString
  /**
   * es distribution flavor
   * @aliases f
   */
  'flavor'?: string
  /**
   * es distribution type
   * @aliases t
   */
  'type'?: Type
  /**
   * es build hash
   * @aliases b
   */
  'build'?: string
  /**
   * jdk version
   * @aliases j
   */
  'jdk'?: string
  /**
   * total disk space
   * @aliases dt,diskTotal
   */
  'disk.total'?: ByteSize
  /**
   * used disk space
   * @aliases du,diskUsed
   */
  'disk.used'?: ByteSize
  /**
   * available disk space
   * @aliases d,da,disk,diskAvail
   */
  'disk.avail'?: ByteSize
  /**
   * used disk space percentage
   * @aliases dup,diskUsedPercent
   */
  'disk.used_percent'?: Percentage
  /**
   * used heap
   * @aliases hc,heapCurrent
   */
  'heap.current'?: string
  /**
   * used heap ratio
   * @aliases hp,heapPercent
   */
  'heap.percent'?: Percentage
  /**
   * max configured heap
   * @aliases hm,heapMax
   */
  'heap.max'?: string
  /**
   * used machine memory
   * @aliases rc,ramCurrent
   */
  'ram.current'?: string
  /**
   * used machine memory ratio
   * @aliases rp,ramPercent
   */
  'ram.percent'?: Percentage
  /**
   * total machine memory
   * @aliases rn,ramMax
   */
  'ram.max'?: string
  /**
   * used file descriptors
   * @aliases fdc,fileDescriptorCurrent
   */
  'file_desc.current'?: string
  /**
   * used file descriptor ratio
   * @aliases fdp,fileDescriptorPercent
   */
  'file_desc.percent'?: Percentage
  /**
   * max file descriptors
   * @aliases fdm,fileDescriptorMax
   */
  'file_desc.max'?: string

  /**
   * recent cpu usage
   */
  'cpu'?: string
  /**
   * 1m load avg
   */
  'load_1m'?: string
  /**
   * 5m load avg
   */
  'load_5m'?: string
  /**
   * 15m load avg
   * @aliases l
   */
  'load_15m'?: string
  /**
   * node uptime
   * @aliases u
   */
  'uptime'?: string
  /**
   * m:master eligible node, d:data node, i:ingest node, -:coordinating node only
   * @aliases r,role,nodeRole
   */
  'node.role'?: string

  /**
   * *:current master
   * @aliases m
   */
  'master'?: string

  /**
   * node name
   * @aliases n
   */
  'name'?: Name

  /**
   * size of completion
   * @aliases cs,completionSize
   */
  'completion.size'?: string

  /**
   * used fielddata cache
   * @aliases fm,fielddataMemory
   */
  'fielddata.memory_size'?: string

  /**
   * fielddata evictions
   * @aliases fe,fielddataEvictions
   */
  'fielddata.evictions'?: string

  /**
   * used query cache
   * @aliases qcm,queryCacheMemory
   */
  'query_cache.memory_size'?: string

  /**
   * query cache evictions
   * @aliases qce,queryCacheEvictions
   */
  'query_cache.evictions'?: string

  /**
   * query cache hit counts
   * @aliases qchc,queryCacheHitCount
   */
  'query_cache.hit_count'?: string

  /**
   * query cache miss counts
   * @aliases qcmc,queryCacheMissCount
   */
  'query_cache.miss_count'?: string

  /**
   * used request cache
   * @aliases rcm,requestCacheMemory
   */
  'request_cache.memory_size'?: string

  /**
   * request cache evictions
   * @aliases rce,requestCacheEvictions
   */
  'request_cache.evictions'?: string

  /**
   * request cache hit counts
   * @aliases rchc,requestCacheHitCount
   */
  'request_cache.hit_count'?: string

  /**
   * request cache miss counts
   * @aliases rcmc,requestCacheMissCount
   */
  'request_cache.miss_count'?: string

  /**
   * number of flushes
   * @aliases ft,flushTotal
   */
  'flush.total'?: string

  /**
   * time spent in flush
   * @aliases ftt,flushTotalTime
   */
  'flush.total_time'?: string

  /**
   * number of current get ops
   * @aliases gc,getCurrent
   */
  'get.current'?: string

  /**
   * time spent in get
   * @aliases gti,getTime
   */
  'get.time'?: string

  /**
   * number of get ops
   * @aliases gto,getTotal
   */
  'get.total'?: string

  /**
   * time spent in successful gets
   * @aliases geti,getExistsTime
   */
  'get.exists_time'?: string

  /**
   * number of successful gets
   * @aliases geto,getExistsTotal
   */
  'get.exists_total'?: string

  /**
   * time spent in failed gets
   * @aliases gmti,getMissingTime
   */
  'get.missing_time'?: string

  /**
   * number of failed gets
   * @aliases gmto,getMissingTotal
   */
  'get.missing_total'?: string

  /**
   * number of current deletions
   * @aliases idc,indexingDeleteCurrent
   */
  'indexing.delete_current'?: string

  /**
   * time spent in deletions
   * @aliases idti,indexingDeleteTime
   */
  'indexing.delete_time'?: string

  /**
   * number of delete ops
   * @aliases idto,indexingDeleteTotal
   */
  'indexing.delete_total'?: string

  /**
   * number of current indexing ops
   * @aliases iic,indexingIndexCurrent
   */
  'indexing.index_current'?: string

  /**
   * time spent in indexing
   * @aliases iiti,indexingIndexTime
   */
  'indexing.index_time'?: string

  /**
   * number of indexing ops
   * @aliases iito,indexingIndexTotal
   */
  'indexing.index_total'?: string

  /**
   * number of failed indexing ops
   * @aliases iif,indexingIndexFailed
   */
  'indexing.index_failed'?: string

  /**
   * number of current merges
   * @aliases mc,mergesCurrent
   */
  'merges.current'?: string

  /**
   * number of current merging docs
   * @aliases mcd,mergesCurrentDocs
   */
  'merges.current_docs'?: string

  /**
   * size of current merges
   * @aliases mcs,mergesCurrentSize
   */
  'merges.current_size'?: string

  /**
   * number of completed merge ops
   * @aliases mt,mergesTotal
   */
  'merges.total'?: string

  /**
   * docs merged
   * @aliases mtd,mergesTotalDocs
   */
  'merges.total_docs'?: string
  /**
   * size merged
   * @aliases mts,mergesTotalSize
   */
  'merges.total_size'?: string
  /**
   * time spent in merges
   * @aliases mtt,mergesTotalTime
   */
  'merges.total_time'?: string

  /**
   * total refreshes
   */
  'refresh.total'?: string
  /**
   * time spent in refreshes
   */
  'refresh.time'?: string
  /**
   * total external refreshes
   * @aliases rto,refreshTotal
   */
  'refresh.external_total'?: string
  /**
   * time spent in external refreshes
   * @aliases rti,refreshTime
   */
  'refresh.external_time'?: string
  /**
   * number of pending refresh listeners
   * @aliases rli,refreshListeners
   */
  'refresh.listeners'?: string

  /**
   * script compilations
   * @aliases scrcc,scriptCompilations
   */
  'script.compilations'?: string
  /**
   * script cache evictions
   * @aliases scrce,scriptCacheEvictions
   */
  'script.cache_evictions'?: string
  /**
   * script cache compilation limit triggered
   * @aliases scrclt,scriptCacheCompilationLimitTriggered
   */
  'script.compilation_limit_triggered'?: string

  /**
   * current fetch phase ops
   * @aliases sfc,searchFetchCurrent
   */
  'search.fetch_current'?: string
  /**
   * time spent in fetch phase
   * @aliases sfti,searchFetchTime
   */
  'search.fetch_time'?: string
  /**
   * total fetch ops
   * @aliases sfto,searchFetchTotal
   */
  'search.fetch_total'?: string
  /**
   * open search contexts
   * @aliases so,searchOpenContexts
   */
  'search.open_contexts'?: string
  /**
   * current query phase ops
   * @aliases sqc,searchQueryCurrent
   */
  'search.query_current'?: string
  /**
   * time spent in query phase
   * @aliases sqti,searchQueryTime
   */
  'search.query_time'?: string
  /**
   * total query phase ops
   * @aliases sqto,searchQueryTotal
   */
  'search.query_total'?: string
  /**
   * open scroll contexts
   * @aliases scc,searchScrollCurrent
   */
  'search.scroll_current'?: string
  /**
   * time scroll contexts held open
   * @aliases scti,searchScrollTime
   */
  'search.scroll_time'?: string
  /**
   * completed scroll contexts
   * @aliases scto,searchScrollTotal
   */
  'search.scroll_total'?: string

  /**
   * number of segments
   * @aliases sc,segmentsCount
   */
  'segments.count'?: string
  /**
   * memory used by segments
   * @aliases sm,segmentsMemory
   */
  'segments.memory'?: string
  /**
   * memory used by index writer
   * @aliases siwm,segmentsIndexWriterMemory
   */
  'segments.index_writer_memory'?: string
  /**
   * memory used by version map
   * @aliases svmm,segmentsVersionMapMemory
   */
  'segments.version_map_memory'?: string
  /**
   * memory used by fixed bit sets for nested object field types  and export type filters for types referred in _parent fields
   * @aliases sfbm,fixedBitsetMemory
   */
  'segments.fixed_bitset_memory'?: string

  /**
   * number of current suggest ops
   * @aliases suc,suggestCurrent
   */
  'suggest.current'?: string
  /**
   * time spend in suggest
   * @aliases suti,suggestTime
   */
  'suggest.time'?: string
  /**
   * number of suggest ops
   * @aliases suto,suggestTotal
   */
  'suggest.total'?: string

  /**
   * number of bulk shard ops
   * @aliases bto,bulkTotalOperations
   */
  'bulk.total_operations'?: string
  /**
   * time spend in shard bulk
   * @aliases btti,bulkTotalTime
   */
  'bulk.total_time'?: string
  /**
   * total size in bytes of shard bulk
   * @aliases btsi,bulkTotalSizeInBytes
   */
  'bulk.total_size_in_bytes'?: string
  /**
   * average time spend in shard bulk
   * @aliases bati,bulkAvgTime
   */
  'bulk.avg_time'?: string
  /**
   * average size in bytes of shard bulk
   * @aliases basi,bulkAvgSizeInBytes
   */
  'bulk.avg_size_in_bytes'?: string
}
